File: net/src/main/java/com/wgh/mvpframework/net/api/TestApi.java
package com.wgh.mvpframework.net.api;

import com.wgh.mvpframework.net.model.TestBean;
import com.wgh.mvpframework.net.model.TestModel;

import io.reactivex.Observable;
import retrofit2.http.Body;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;

/**
 * @version V0.1.0
 * @author: wgh
 * @date: 2018/4/10
 * @description
 */
public interface TestApi {

    @GET("get")
    Observable<TestBean> testGet();

    @GET("data/{page}")
    Observable<TestBean> testGet(@Path("page") int page);

    @GET("get")
    Observable<TestBean> testGet(@Query("username") String username, @Query("pwd") String pwd);

    @FormUrlEncoded
    @POST("post")
    Observable<TestBean> testPost(@Field("username") String username, @Field("pwd") String pwd);

    @POST("post")
    Observable<TestBean> testPost(@Body TestModel body);


}

File: app/src/main/java/com/wgh/mvpframework/network/api/TestApi.kt
package com.wgh.mvpframework.network.api

import com.wgh.mvpframework.base.BaseBean
import io.reactivex.Observable
import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.QueryMap

/**
 * @author: wgh
 * @date: 2018/4/3
 * @version V0.1.0
 * @description
 */
interface TestApi {

    @POST("post")
    fun test(@QueryMap options: Map<String, String>) : Observable<String>

    @POST("post")
    fun <T : BaseBean> test(@Body bean: Class<T>) : Observable<String>

}